utils_linux.go

```diff
@@ -441,7 +441,7 @@ func startContainer(context *cli.Context, spec *specs.Spec, action CtAct, criuOp
 
 	r := &runner{
 		enableSubreaper: !context.Bool("no-subreaper"),
-		shouldDestroy:   true,
+		shouldDestroy:   !context.Bool("no-delete"),
```

Comment:
Not something new in this PR (so don't change) but it would be good to use the "default" values match the default behavior. (e.g. change the field to `keep` or `disableAutoDestroy`), so that the default for "boolean" values (`false`) also is the expected "default"

Reply:
Yes, it dose look strange to treat the default value `false` of `--no-delete` as auto-delete the container.
May be we can change `--no-delete` to `--disable-autodestroy`?

Reply:
I have suggested `--keep-cgroup` in #2817, but as we also have to keep the runc state directory, maybe just `--keep`?

Reply:
Yes, I was considering "--keep" as well; so sgtm

Reply:
SGTM too.
Welcome more better suggestions...

Reply:
The reason I suggested `--no-delete` was because it is consistent with the three existing `--no-*` flags in `runc run` and `delete` is consistent with the existing `runc delete` command that has the matching behavior. As a user, I don't care of course, and can use any flag.

Reply:
The existing `--no-*` flags are
```
   --no-subreaper            disable the use of the subreaper used to reap reparented processes
   --no-pivot                do not use pivot root to jail process inside rootfs.  This should be used whenever the rootfs is on top of a ramdisk
   --no-new-keyring          do not create a new session keyring for the container.  This will cause the container to inherit the calling processes session key
```

and in all 3 cases it's a `--no-NOUN`. There's also "--preserve-fds`...

To me `--keep` still looks like the best choice, the only issue is it's too generic (keep everything?).

Reply:
I guess I wouldn't consider `pivot` noun in this context (short for "pivot root"). Any flag is ok for me though.

Reply:
The difference is that for all of those `--no` settings, you're disabling a security feature and so the default should be false. Since the default for this option should also be false, it should be named something that makes sense in that context.